src/utils/utils.js

```diff
@@ -0,0 +1,26 @@
+export default function timeDifference(dateString) {
+    const currentDate = new Date();
+    const createdDate = new Date(dateString);
+  
+    const timeDifference = currentDate - createdDate;
+    const seconds = Math.floor(timeDifference / 1000);
+    const minutes = Math.floor(seconds / 60);
+    const hours = Math.floor(minutes / 60);
+    const days = Math.floor(hours / 24);
+    const months = Math.floor(days / 30);
+    const years = Math.floor(months / 12);
+  
+    if (years > 0) {
+      return `${years} ${years === 1 ? 'year' : 'years'} ago`;
+    } else if (months > 0) {
+      return `${months} ${months === 1 ? 'month' : 'months'} ago`;
+    } else if (days > 0) {
+      return `${days} ${days === 1 ? 'day' : 'days'} ago`;
+    } else if (hours > 0) {
+      return `${hours} ${hours === 1 ? 'hour' : 'hours'} ago`;
+    } else if (minutes > 0) {
+      return `${minutes} ${minutes === 1 ? 'minute' : 'minutes'} ago`;
+    } else {
+      return `${seconds} ${seconds === 1 ? 'second' : 'seconds'} ago`;
+    }
+  }
```

Comment:
There are libraries that could do this but really like the creativity...however - if you're going to have utils like this I would probably have tested it

Reply:
I'll have a look at these libraries. It makes sense to test for utils like this one.